Add spec for governance token staking from general account and locked rewards counting towards staking balance (#2350)
vegaprotocol/specs: protocol/0013-ACCT-accounts.md

## Accounts controlled by parties

A party only has control over balances in the "general" and "LOCKED_FOR_STAKING" account for each asset.
[Parties](./0017-PART-party.md) are identified by Vega public keys. Each party that makes a deposit on one of the asset bridges, currently only [Ethereum ERC20 bridge](./0031-ETHB-ethereum_bridge_spec.md) will have a general account for the relevant [asset](./0040-ASSF-asset_framework.md) created.

> Each party that has locked and vesting rewards will have a "LOCKED_FOR_STAKING" account for the relevant [asset] created, party can transfer the tokens in and out of this LOCKED_FOR_STAKING account. The balance in this account should count towards the tokens [associated](./0059-STKG-simple_staking_and_delegating.md) with the Vega key for staking purposes and can be staked to specific validators, earning staking rewards for that key.

Reviewer: I believe that the funds will be staying in the vesting / vested accounts, LOCKED_FOR_STAKING should be used only from a general account. 

The account will be created on demand if needed.

> A party can transfer governance tokens in and out of the "LOCKED_FOR_STAKING" account. The balance in this account should count towards the tokens "associated" with the Vega key for staking purposes and can be staked to specific validators, earning staking rewards for that key.

Reviewer 2: ```suggestion
A party can transfer governance tokens in and out of the "LOCKED_FOR_STAKING" account. The balance in this account should count towards the tokens [associated](./0059-STKG-simple_staking_and_delegating.md) with the Vega key for staking purposes and can be staked to specific validators, earning staking rewards for that key.
```

## Accounts controlled by parties

A party only has control over balances in the "general" and "LOCKED_FOR_STAKING" account for each asset.

> [Parties](./0017-PART-party.md) are identified by Vega public keys. Each party that makes a deposit on one of the asset bridges, currently only [Ethereum ERC20 bridge](./0031-ETHB-ethereum_bridge_spec.md) will have a general account and "LOCKED_FOR_STAKING" for the relevant [asset](./0040-ASSF-asset_framework.md) created.

Reviewer 2: ```suggestion
[Parties](./0017-PART-party.md) are identified by Vega public keys. Each party that makes a deposit on one of the asset bridges, currently only [Ethereum ERC20 bridge](./0031-ETHB-ethereum_bridge_spec.md) will have a general account for the relevant [asset](./0040-ASSF-asset_framework.md) created.
A party that transfers any balance of the governance token into  its "LOCKED_FOR_STAKING" account will have that account created.
```

A party can transfer governance tokens in and out of the "LOCKED_FOR_STAKING" account. The balance in this account should count towards the tokens [associated](./0059-STKG-simple_staking_and_delegating.md) with the Vega key for staking purposes and can be staked to specific validators, earning staking rewards for that key.

> Vega core listens to events on other chains (currently Ethereum) for events on the bridge contract. If governance token reward is locked on the bridge contract with a Vega party key associated then the [Locked_FOR_REWARD](./0013-ACCT-accounts.md) balance for the party is incremented by the appropriate amount.

Reviewer 2: ```suggestion
```

- It is possible to deposit funds from Ethereum directly into the global rewards account by specifying the `0` Vega address. (<a name="0013-ACCT-027" href="#0013-ACCT-027">0013-ACCT-027</a>)
- It is possible to transfer funds from a Vega general account to the global rewards account by specifying the `0` address and appropriate account type. (<a name="0013-ACCT-028" href="#0013-ACCT-028">0013-ACCT-028</a>)

> 

Reviewer 2: AC for rewards are in a different place, I wouldn't randomly add them here.
